### PR TITLE
[ParallelCluster-API] Add missing policies for EcrImageDeletionLambda and ImageBuilderInstance Roles

### PR DESCRIPTION
**Pull request by EddyMM**
### Description of changes
* CloudFormation Stack upgrade of the ParallelCluster API fails for ECRImage and ECRImageRemover resources due to missing actions and policies.
* This commit adds the missing policies & actions to the corresponding roles

### Tests
* Manually tested by:
  * Reproducing the issue with existing API template
  * Applying the fixes in this commit
  * Upgrading the API Version using `aws cloudformation update-stack` as instructed in the [ParallelCluster API Docs](https://docs.aws.amazon.com/parallelcluster/latest/ug/api-reference-v3.html#api-reference-update-v3)
  * Confirmed the cluster is successfully updated

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: api/infrastructure/parallelcluster-api.yaml [hunk 1]
```diff
@@ -964,6 +964,17 @@ Resources:
...
         - !Sub arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore
         - !Sub arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilderECRContainerBuilds
+      Policies:
+        - PolicyName: ImageBuilderCustomPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: ecs:RegisterContainerInstance
+                Resource: !Sub arn:${AWS::Partition}:ecs:*:${AWS::AccountId}:cluster/*
```
Comment (gmarciani) [inline api/infrastructure/parallelcluster-api.yaml:974]: Is it possible to scope the permission down by restricting more the resource pattern, e.g. including the StackIdSuffix?

File: api/infrastructure/parallelcluster-api.yaml [hunk 2]
```diff
@@ -964,6 +964,17 @@ Resources:
...
+              - Effect: Allow
+                Action: ecs:DiscoverPollEndpoint
+                Resource: "*"
```
Comment (gmarciani) [inline api/infrastructure/parallelcluster-api.yaml:977]: Is it possible to scope the permission down by restricting more the resource pattern, e.g. including the StackIdSuffix or at least restricting to specific resources owned by the account on the targeted region?

File: api/infrastructure/parallelcluster-api.yaml (hunk 1 above)
Comment (EddyMM) [reply api/infrastructure/parallelcluster-api.yaml:974]: The `RegisterContainerInstance` API Call targets a cluster that doesn't include a StackIdSuffix. The CloudTrail Log shows `cluster/default` so this is the closest resource pattern we can get without being too restrictive.

File: api/infrastructure/parallelcluster-api.yaml (hunk 2 above)
Comment (EddyMM) [reply api/infrastructure/parallelcluster-api.yaml:977]: Similarly, the `DiscoverPollEndpoint` API Call is done against a `*` resource based on the CloudTrail Logs.

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,6 +40,7 @@ CHANGELOG
 - Fix default for disable validate and test components when building custom AMI. The default was to disable those components, but it wasn't effective.
 - Handle corner case in the scaling logic when instance is just launched and the describe instances API doesn't report yet all the EC2 info.
 - Dropped validation that would prevent ARM instance type to be used when `DisableSimultaneousMultithreading` was set to true.
+- Fix stack update failure when upgrading ParallelCluster API from one version to another.
```
Comment (gmarciani) [inline CHANGELOG.md]: What about "Add missing policies for EcrImageDeletionLambda and ImageBuilderInstance roles that were causing failure when upgrading ParallelCluster API from one version to another" ?

File: api/infrastructure/parallelcluster-api.yaml (hunk 2 above)
Comment (gmarciani) [reply api/infrastructure/parallelcluster-api.yaml:977]: What do you mean with "is done against *"? Do you mean that the resource name does not follow any specific pre-determined pattern?

Comment (EddyMM) [reply api/infrastructure/parallelcluster-api.yaml:977]: The CloudTrail logs don't specify which resource the role is unable to make a `DiscoverPollEndpoint` call to.

```
    "errorMessage": "User: arn:...ImageBuilderInstanceRole-... is not authorized to perform: ecs:DiscoverPollEndpoint on resource: * because no identity-based policy allows the ecs:DiscoverPollEndpoint action",
```

Comment (gmarciani) [reply api/infrastructure/parallelcluster-api.yaml:977]: Ok, so let's go with this implementation for this iteration.
By the way, let's be sure to ask ECS suggestions on how to scope it down

File: api/infrastructure/parallelcluster-api.yaml (hunk 1 above)
Comment (gmarciani) [reply api/infrastructure/parallelcluster-api.yaml:974]: Let's go with this implementation for this iteration.
By the way, let's be sure to ask ECS suggestions on how to scope it down.
For instance, what about considering (**not in this PR**) to scope it down using conditions on tags?

**Review by gmarciani — APPROVED** [review]
